### PR TITLE
[Plugin|tags] Add tagging to numerous plugins for Insights

### PR DESCRIPTION
**Pull request by TurboTurtle**
This patchset adds tags to several plugins for aiding the Insights project when it analyzes provided sos archives.

Resolves: #2536
---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [x] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [x] Is the subject and message clear and concise?
- [x] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [x] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [ ] If this commit closes an existing issue, is the line `Closes: #ISSUENUMBER` included in an independent line?
- [x] If this commit resolves an existing pull request, is the line `Resolves: #PRNUMBER` included in an independent line?

### CODE REVIEW

File: sos/report/plugins/__init__.py [hunk 1]
```diff
@@ -1516,6 +1516,8 @@ def time_filter(path):
                     # in the corner case we just reached the sizelimit, we
                     # should collect the whole file and stop
                     limit_reached = (sizelimit and current_size == sizelimit)
+
+            _spec_tags = list(set(_spec_tags))
```
Comment (pmoravec) [inline sos/report/plugins/__init__.py:1520]: Why this list unification? (well, it doesnt make sense to have a tag multiple times - but then we can use a set type directly..?)

Comment (TurboTurtle) [reply sos/report/plugins/__init__.py:1520]: Yeah, it's just to scrub duplicates. The various tag-adding bits allow for strings or lists of strings, so doing the reduction here I feel is easier than transitioning the methods to using sets directly.

File: sos/report/plugins/__init__.py [hunk 2]
```diff
@@ -1912,6 +1922,16 @@ def _collect_cmd_output(self, cmd, suggest_filename=None,
         _tags.append(cmd.split(' ')[0])
         _tags.extend(self.get_tags_for_cmd(cmd))
 
+        if cmd_as_tag:
+            _cmdtag = cmd
+            for s in (' ', '/', '-'):
+                _cmdtag = _cmdtag.replace(s, '_')
+            # reduce duplicates to a single underscore
+            _cmdtag = _cmdtag.replace('__', '_')
```
Comment (pmoravec) [inline sos/report/plugins/__init__.py:1930]: Can't we use `_mangle_command` from https://github.com/sosreport/sos/blob/master/sos/report/plugins/__init__.py#L36, as being used for `outfn` in https://github.com/sosreport/sos/blob/master/sos/report/plugins/__init__.py#L1803 ? Then tag would match the `outfn` filename.

File: sos/report/plugins/foreman.py [hunk 3]
```diff
@@ -326,6 +334,12 @@ class RedHatForeman(Foreman, SCLPlugin, RedHatPlugin):
     apachepkg = 'httpd'
 
     def setup(self):
+
+        self.add_file_tags({
+            '/var/log/foreman-installer/satellite.log.*': 'satellite_log',
```
Comment (pmoravec) [inline sos/report/plugins/foreman.py:339]: Maybe tagname `insataller_satellite_log` or `satellite_installer_log`? As under the term "the" Satellite log, I would imagine rather `/var/log-foreman-production.log`.

File: sos/report/plugins/processor.py [hunk 4]
```diff
@@ -19,6 +19,18 @@ class Processor(Plugin, IndependentPlugin):
     packages = ('cpufreq-utils', 'cpuid')
 
     def setup(self):
+
+        cpupath = '/sys/devices/system/cpu'
+
+        self.add_file_tags({
+            "%s/smt/control" % cpupath: 'cpu_smt_control',
+            "%s/smt/active": 'cpu_smt_active',
```
Comment (pmoravec) [inline sos/report/plugins/processor.py:27]: ` % cpupath` is missing here.

File: sos/report/plugins/processor.py [hunk 5]
```diff
@@ -19,6 +19,18 @@ class Processor(Plugin, IndependentPlugin):
...
+            "%s/vulnerabilities/.*" % cpupath: 'cpu_vulns',
+            "%s/vulnerabilities/spectre_v2" % cpupath: 'cpu_vulns_spectre_v2',
+            "%s/vulnerabilities/meltdown" % cpupath: 'cpu_vulns_meltdown',
+            "%s/cpu.*/online": 'cpu_cores'
```
Comment (pmoravec) [inline sos/report/plugins/processor.py:31]: .. and also missing it here :)

File: sos/report/plugins/__init__.py (hunk 2 above)
Comment (TurboTurtle) [reply sos/report/plugins/__init__.py:1930]: Hadn't thought of that, but yeah that makes sense.

File: sos/report/plugins/foreman.py (hunk 3 above)
Comment (TurboTurtle) [reply sos/report/plugins/foreman.py:339]: This was to match the name Insights uses for it already. We could do `insights_satellite_log` and `satellite_installer_log` though just as easily.

**Review by pmoravec — APPROVED** [review]

**Pull request merged**